### PR TITLE
TRUNK-6047: Add utility methods to invalidate database caches

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6047

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by IamMujuziMoses** [comment]
@wluyima @dkayiwa @ibacher could you please review this PR

File: api/src/main/java/org/openmrs/api/context/Context.java [hunk 1]
```diff
@@ -890,6 +892,57 @@ public static void evictFromSession(Object obj) {
...
 	}
 
+	/**
+	 * Evicts the entity data for a particular entity instance.
+	 *
+	 * @param sessionFactory
+	 * @param entityClass entity class of the instance to evict from the DB cache
+	 * @param object entity instance to evict from the DB cache
+	 */
+	public static void evictSingleEntity(SessionFactory sessionFactory, Class<?> entityClass, OpenmrsObject object) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/api/context/Context.java:902]: I would model this on how `evictFromSession()` above works. TBH, I'm not actually sure why we need this.

File: api/src/main/java/org/openmrs/api/context/Context.java [hunk 2]
```diff
@@ -890,6 +892,57 @@ public static void evictFromSession(Object obj) {
...
+		
+		String uuid = object != null ? object.getUuid() : null;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/api/context/Context.java:904]: This is a weird thing to do... It makes sense in the REST module, which largely operates on UUIDs, but not here.

**Review by ibacher — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/api/context/Context.java (hunk 1 above)
Comment (IamMujuziMoses) [reply api/src/main/java/org/openmrs/api/context/Context.java:902]: Using ```evictFromSession()``` doesn't seem to do the trick of removing a single entity from cache as shown [here](https://drive.google.com/file/d/1fDt0VemSFk5CteqhdTlMZh1zBSWBo4E5/view?usp=sharing).

**Comment by IamMujuziMoses** [comment]
I accidentally deleted this head branch in my forked repo leading to the closing of this PR, and I couldn't reopen it so I have created a new [PR](https://github.com/openmrs/openmrs-core/pull/4421), sorry for thee inconveniences 😧

**Pull request closed without merging**